gradio/blocks.py

```diff
@@ -1046,3 +1056,17 @@ def block_thread(
             self.server.close()
             if self.enable_queue:
                 queueing.close()
+
+    def configure_random_values(self):
+        """Add a load event for every component whose initial value should be randomized."""
+        for component in Context.root_block.blocks.values():
```

Comment:
So this approach works, but I think there's a better way that doesn't involve iterating over all of the blocks (many of whom don't need to know about the concept of randomization). 

What if we move this logic to the `render()` method of the `IOComponent` class? The `IOComponent.render()` would first call `super().render()`  and include this logic: if `component.randomize`, then look up the parent block (using `Context.root_block`) and create an event trigger

Reply:
I like it! Much more targeted change since it only applies to components that can take on values. 

Will give this a shot and let you know if I have questions!

Reply:
If we supported custom functions, it would be a simple change to swap out randomize to swap out `if component.randomize` for `if component.value isinstance Callable`. Or even support both, such as

```python
load_fn = None
if component.randomize:
  load_fn = component.get_random_value_on_load
elif component.value isinstance Callable:
  load_fn = component.value
if load_fn:
self.set_event_trigger(
      "load",
      load_fn,
      None,
      component,
      no_target=True,
      queue=False,
  )
```

Reply:
This is now implemented @aliabid94 @abidlabs